pkg/controllers/datastoresyncer/datastore_cluster_sync_test.go

```diff
@@ -112,3 +115,37 @@ func testClusterSyncing() {
 		})
 	})
 }
+
+func testClusterCleanup() {
+	t := newTestDriver()
+
+	BeforeEach(func() {
+		t.doStart = false
+
+		test.CreateResource(t.localClusters, newCluster(&t.localCluster.Spec))
+		test.CreateResource(t.brokerClusters, newCluster(&t.localCluster.Spec))
+
+		test.CreateResource(t.localClusters, newCluster(&submarinerv1.ClusterSpec{
+			ClusterID: otherClusterID,
+		}))
+		test.CreateResource(t.brokerClusters, newCluster(&submarinerv1.ClusterSpec{
+			ClusterID: otherClusterID,
+		}))
+	})
+
+	It("should remove local Clusters from the remote datastore", func() {
+		Expect(t.syncer.Cleanup()).To(Succeed())
+
+		test.AwaitNoResource(t.brokerClusters, clusterID)
+
+		time.Sleep(500 * time.Millisecond)
+		test.AwaitResource(t.brokerClusters, otherClusterID)
```

Comment:
Usually every cluster has only a single Cluster Object right, any specific reason for validating two cluster objects? 
To be more precise, if we are not the owner for otherClusterID, why are we expecting that to be deleted on the Broker?

Reply:
Oh yeah. Indeed i overlooked it :P